src/components/Accordion/Accordion.tsx

```diff
@@ -0,0 +1,141 @@
+import _ from 'lodash';
+import React from 'react';
+import PropTypes from 'react-peek/prop-types';
+import { lucidClassNames } from '../../util/style-helpers';
+import { buildHybridComponent } from '../../util/state-management';
+import {
+	ExpanderPanelDumb as ExpanderPanel,
+	IExpanderPanelHeaderProps,
+} from '../ExpanderPanel/ExpanderPanel';
+import {
+	findTypes,
+	omitProps,
+	StandardProps,
+	Overwrite,
+} from '../../util/component-types';
+import * as reducers from '../Accordion/Accordion.reducers';
+
+const cx = lucidClassNames.bind('&-Accordion');
+
+const { any, func, object, number, string } = PropTypes;
+
+interface IAccordionPropsRaw extends StandardProps {
+	/**
+	 * Indicates which item is expanded
+	 * */
+	selectedIndex: number;
+	/**
+	 * Called when the user clicks on the component's header of an item.
+	 * */
+
+	onSelect: (
+		selectedIndex: number | null,
+		{ event, props }: { event: React.MouseEvent; props: IAccordionProps }
+	) => void;
+	/**
+	 * Prop alternative to Header child component passed through to the
+	 * underlying ExpanderPanel
+	 */
+	Header: IExpanderPanelHeaderProps;
+}
+
+type IAccordionProps = Overwrite<
+	React.DetailedHTMLProps<React.HTMLAttributes<HTMLDivElement>, HTMLDivElement>,
+	IAccordionPropsRaw
+>;
+
+export interface IAccordionState {
+	selectedIndex: number | null;
+}
+
+const defaultProps = {
+	onSelect: _.noop,
+};
+
+class Accordion extends React.Component<IAccordionProps, IAccordionState> {
+	static displayName = 'Accordion';
+	static propTypes = {
+		className: string`
+			Appended to the component-specific class names set on the root element.
+		`,
+		selectedIndex: number`
+			Indicates which item is expanded
+		`,
+		onSelect: func`
+			Called when the user clicks on the component's header of an item.
+		`,
+		style: object`
+			Passed through to the root element.
+		`,
+		Header: any`
+			Prop alternative to Header child component passed through to the
+			underlying ExpanderPanel
+		`,
+	};
+	static definition = {
+		statics: {
+			Item: ExpanderPanel,
+			Header: ExpanderPanel.Header,
+			reducers,
+			peek: {
+				description: `
+					This is a container that renders panels and controls its
+					expansion/retraction.
+				`,
+				categories: ['layout'],
+				madeFrom: ['ExpanderPanel'],
+			},
+		},
+	};
+
+	static defaultProps = defaultProps;
+
+	static reducers = reducers;
+
+	static Item = ExpanderPanel;
+
+	static Header = ExpanderPanel.Header;
+
+	handleToggle(isExpanded: boolean, index: number, event: React.MouseEvent) {
+		const selectedIndex = isExpanded ? index : null;
+
+		this.props.onSelect(selectedIndex, {
+			event,
+			props: this.props,
+		});
+	}
+
+	render() {
+		const { style, className, selectedIndex, ...passThroughs } = this.props;
+
+		const itemChildProps = _.map(
+			findTypes(this.props, Accordion.Item as { propName?: string | string[] }),
```

Comment:
I'm not sure why this needs to be type asserted. Just passing `Accordion.Item` doesn't type check. A lot of components use `findTypes` without the type assertion so I would assume it should work.

Reply:
error for reference
```
Argument of type 'Readonly<Overwrite<DetailedHTMLProps<HTMLAttributes<HTMLDivElement>, HTMLDivElement>, IAccordionPropsRaw>> & Readonly<{ children?: ReactNode; }>' is not assignable to parameter of type 'IExpanderPanelProps'.
```

Reply:
Thinking about this some more, I think maybe this pattern should not be using ExpanderPanel directly. Like how CalendarMonth gets used here: https://github.com/appnexus/lucid/blob/5ab4bb2fb861fd38c92e12e07f46d1fbbdef735f/src/components/DateSelect/DateSelect.tsx.

So Accordian.Item should not actually be an ExpanderPanel and should just be some dummy like
```
const AccordianItem = (_props: IAccordianItemProps): null =>
	null;
```

What do you think?

Reply:
I think you're right that using a dummy child component would work. I'm not sure it's necessary as long as we're getting the right type signatures for the consumers. I.e. might not be worth the effort/risk if this is working okay with a type cast.